src/App.tsx

```diff
@@ -1,9 +1,32 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
+import moviesFromServer from './api/movies.json';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
-import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const getFilteredMovies = () => {
+    if (!query) {
+      return moviesFromServer;
+    }
+
+    const isMatch
+    = (text: string,
+      textQuery: string) => text.toLowerCase().includes(textQuery);
+
+    const lowerCaseQuery = query.toLowerCase().trim();
+
+    return moviesFromServer.filter(movie => isMatch(movie.title, lowerCaseQuery)
+      || isMatch(movie.description, lowerCaseQuery));
+  };
+
+  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => (
```

Comment:
```suggestion
  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => (
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364